esi/lib/nodes.py

```diff
@@ -138,3 +140,95 @@ def network_list(connection, filter_node=None, filter_network=None):
             })
 
     return data
+
+
+def network_attach(connection, node, **kwargs):
```

Comment:
I think I do prefer the network information being passed in as a dictionary, similar to https://opendev.org/openstack/metalsmith/src/branch/master/metalsmith/_provisioner.py#L335

Reply:
Would it be a list of dictionaries? Because there can only be one of network, port, and trunk. Only mac-address can have multiple

Reply:
No, just something like:

```
{
   "network": "network uuid",
   "port": "port_uuid",
}
```

And then the sdk function would validate the contents of the dictionary and error out if both network and port are defined.